### PR TITLE
docs(python): Add note on hash stability and mention `polars-hash`

### PR DESCRIPTION
**Pull request by ion-elgreco**
As per discussion on discord @MarcoGorelli, adding a note on the instability of `hash` across versions and mentioning they can use polars-hash if they require stability.

### CODE REVIEW

File: py-polars/polars/expr/expr.py [hunk 1]
```diff
@@ -5370,6 +5370,12 @@ def hash(
 
         The hash value is of type `UInt64`.
 
+        .. note::
+            This implementation of :func:`hash` does not guarantee stable results
+            across polars versions. It only has guaranteed stability within one version.
+            For use-cases where you require stability across version you can use
+            the following plugin: `polars-hash <https://github.com/ion-elgreco/polars-hash>`_.
```
Comment (stinodego) [inline py-polars/polars/expr/expr.py:5377]:
```suggestion
        .. note::
            This implementation of :func:`hash` does not guarantee stable results
            across Polars versions. It only has guaranteed stability within one version.
            For use-cases where you require stability across versions, you can use
            the following plugin: `polars-hash <https://github.com/ion-elgreco/polars-hash>`_.
```

**Review by stinodego — CHANGES_REQUESTED** [review]
Could you add this note to `Series.hash` too? And possibly to `DataFrame.hash_rows` (not sure what that one does exactly).

File: py-polars/polars/expr/expr.py [hunk 2]
```diff
@@ -5370,6 +5370,12 @@ def hash(
 
         The hash value is of type `UInt64`.
 
+        .. note::
```
Comment (stinodego) [inline py-polars/polars/expr/expr.py:5373]: This should be in a `Notes` section below the parameters section.

Comment (ion-elgreco) [reply py-polars/polars/expr/expr.py:5373]: Ah ok didn't know that, I just grabbed a random example from the code and followed that.

**Comment by ion-elgreco** [comment]
Done! @stinodego 

I didn't mention the plugin for hash_rows and Series.hash() since the plugin doesn't have that capability.

File: py-polars/polars/dataframe/frame.py [hunk 3]
```diff
@@ -9485,6 +9485,10 @@ def hash_rows(
         seed_3
             Random seed parameter. Defaults to `seed` if not set.
 
+        .. note::
+            This implementation of :func:`hash_rows` does not guarantee stable results
+            across Polars versions. It only has guaranteed stability within one version.
+
```
Comment (rancomp) [inline py-polars/polars/dataframe/frame.py:9491]:
```suggestion
            This implementation of :func:`hash_rows` does not guarantee consistent results across different Polars versions. Its stability is only guaranteed within a single version.

```
I'd like to suggest this rephrasing for better clarity. What do you think?

**Review by stinodego — APPROVED** [review]

**Comment by stinodego** [comment]
Thanks @ion-elgreco !

File: py-polars/polars/expr/expr.py [hunk 4]
```diff
@@ -5381,6 +5381,14 @@ def hash(
         seed_3
             Random seed parameter. Defaults to `seed` if not set.
 
+        Notes
+        -----
+        This implementation of :func:`rows` does not guarantee stable results
+        across different Polars versions. Its stability is only guaranteed within a
+        single version.
+        For use-cases where stability across versions is required, you can use
```
Comment (ritchie46) [inline py-polars/polars/expr/expr.py:5389]: I think it is good to add the plugin to the list of plugins, but I want to refrain from pointing to plugins in the docstrings we didn't write.

Comment (stinodego) [reply py-polars/polars/expr/expr.py:5389]: All right will revert that part :+1:

**Pull request merged**